src/pubkey.cpp

```diff
@@ -4,19 +4,182 @@
 
 #include "pubkey.h"
 
-#include "eccryptoverify.h"
+#include <secp256k1.h>
+#include <secp256k1_recovery.h>
 
-#include "ecwrapper.h"
+namespace
+{
+secp256k1_context* secp256k1_context_verify = NULL;
```

Comment:
Yes, please not put synchronization in libconsensus, at least until we decide for either C++11 or C for the long term language trend for libconsensus. Last time I checked in IRC, it seems I wasn't alone in preferring a libconsensus that eventually is only C over one that is eventually C++11. Although that decision shouldn't affect this PR or even libconsensus' encapsulation completion, I insist that it would be good very good to know this asap.